src/platform_impl/linux/taskbar/unity.rs

```diff
@@ -0,0 +1,92 @@
+use zbus::{
+  blocking::Connection,
+  fdo::Result,
+  zvariant::{DeserializeDict, SerializeDict, Type},
+  MessageBuilder,
+};
+
+pub struct Manager {
+  conn: Connection,
+  app_uri: String,
+  progress: f64,
+  progress_visible: bool,
+  urgent: bool,
+  dirty_progress: bool,
+  dirty_progress_visible: bool,
+  dirty_urgent: bool,
+}
+
+#[derive(Default, SerializeDict, DeserializeDict, Type, PartialEq, Debug)]
+#[zvariant(signature = "dict")]
+struct Progress {
+  progress: Option<f64>,
+  #[zvariant(rename = "progress-visible")]
+  progress_visible: Option<bool>,
+  urgent: Option<bool>,
+}
+
+impl Manager {
+  pub fn new(app_uri: String) -> Result<Self> {
+    let conn = Connection::session()?;
+    let mut m = Self {
+      conn,
+      app_uri,
+      progress: 0.0,
+      progress_visible: false,
+      urgent: false,
+      dirty_progress: true,
+      dirty_progress_visible: true,
+      dirty_urgent: true,
+    };
+
+    m.update().unwrap_or(());
+    Ok(m)
+  }
+
+  fn update(&mut self) -> Result<()> {
+    let mut properties = Progress::default();
+
+    if self.dirty_progress {
+      self.dirty_progress = false;
+
+      properties.progress = Some(*&self.progress);
```

Comment:
I think `*&` won't be necessary?

Reply:
The compiler complains if not used

Reply:
it shouldn't, `self.progress` is of type `f64` which is a `Copy` type